<!doctype html> <html lang=en> <head> <meta charset=utf-8> <meta content="width=device-width,initial-scale=1" name=viewport> <meta content=#333333 name=theme-color> <link href=/global.css rel=preload as=style onload="this.onload=null;this.rel='stylesheet'"> <link href=//cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/styles/default.min.css rel=preload as=style onload="this.onload=null;this.rel='stylesheet'"> <link href=/manifest.json rel=manifest crossorigin=use-credentials> <link href=/images/favicon.svg rel=icon type=image/svg+xml> <script>__SAPPER__={baseUrl:"",preloaded:[void 0,null,{post:{from:"wordpress",title:"jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処",date:"2016-08-26T00:17:53.000Z",type:"notes",slug:"jedi-virtualenv",html:"\u003Cp\u003Eある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。\npython-modeのバッファでピリオドを打つたびに\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"hljs\"\u003EError (jedi): Failed to start Jedi EPC server.\n*** You may need to run &quot;M-x jedi:install-server&quot;. ***\nThis could solve the problem especially if you haven't run the command yet\nsince Jedi.el installation or update and if the server complains about\nPython module imports.\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003Eというメッセージが出てくるし、当然補完機能は使えない。\n指示通りコマンドを実行しても何も起きない。\u003C\u002Fp\u003E\n\u003C!--more--\u003E\n\u003Cp\u003Eよくみると*Messages*に\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"hljs\"\u003EServer may raise an error. Use &quot;M-x epc:pop-to-last-server-process-buffer RET&quot; to see full traceback:\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003Eとあるのでtracebackを見てみると\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"hljs\"\u003Edyld: Library not loaded: *****\n  Referenced from: *****:\n  Reason: image not found\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003Eという見慣れないエラーがある。なんだこれ。\u003C\u002Fp\u003E\n\u003Ch3\u003E原因\u003C\u002Fh3\u003E\n\u003Cp\u003E簡単に言うとvirtualenvが壊れていた。\n私の環境では \u003Ccode\u003E(custom-set-variables '(jedi:environment-root &quot;jedi&quot;))\u003C\u002Fcode\u003E という\u003Ca href=\"https:\u002F\u002Fgithub.com\u002Fkissge\u002F.emacs.d\u002Fblob\u002Fmaster\u002Finits\u002F15-python.el#L8\"\u003E感じ\u003C\u002Fa\u003Eでvirtualenvを使う設定になっているのだが、そのディレクトリの中にある大量のシンボリックリンクがいずれもリンク切れになっていた。\nhomebrewでPythonをバージョンアップしたためである。\u003C\u002Fp\u003E\n\u003Cp\u003Eと、いうわけで、このディレクトリを削除して作りなおすと万事うまくいった。よかった。\u003C\u002Fp\u003E\n\u003Cp\u003Eあんまし関係無いけど、この問題に対処する隙がない間エラーを無視してとりあえずjediを \u003Ccode\u003Eel-get-remove\u003C\u002Fcode\u003E しようとしたら \u003Ccode\u003Eauto-complete\u003C\u002Fcode\u003E がないから無理〜と言って怒られたのは納得がいかない。\njediさん何とかして。\u003C\u002Fp\u003E\n",headline:"ある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。python-modeのバッファでピリオドを打つたびにError (jedi): Failed to sta"},historyURL:"https:\u002F\u002Fgithub.com\u002Fkissge\u002Fyo.eki.do\u002Fcommits\u002Fmaster\u002Fnotes\u002Fjedi-virtualenv.md"}]};if('serviceWorker' in navigator)navigator.serviceWorker.register('/service-worker.js');(function(){try{eval("async function x(){}");var main="/client/client.ad92a8a1.js"}catch(e){main="/client/legacy/client.8a1ee2e2.js"};var s=document.createElement("script");try{new Function("if(0)import('')")();s.src=main;s.type="module";s.crossOrigin="use-credentials";}catch(e){s.src="/client/shimport@2.0.5.js";s.setAttribute("data-main",main);}document.head.appendChild(s);}());</script> <link href=/client/client-0ed4f0b2.css rel=stylesheet><link href=/client/_slug_-ad5942c7.css rel=stylesheet> <title>jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処 - 2016-08-26 | 葉月夜堂</title><link href=https://twitter.com/p_km rel=me data-svelte=svelte-119a9vg> <link href=https://webmention.io/yo.eki.do/webmention rel=webmention data-svelte=svelte-119a9vg> <link href=https://webmention.io/yo.eki.do/xmlrpc rel=pingback data-svelte=svelte-119a9vg><meta content=article data-svelte=svelte-zqyuod property=og:type><meta content="jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処 - 2016-08-26 | 葉月夜堂" data-svelte=svelte-zqyuod property=og:title><meta content="ある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。python-modeのバッファでピリオドを打つたびにError (jedi): Failed to sta" data-svelte=svelte-zqyuod property=og:description><meta content=葉月夜堂 data-svelte=svelte-zqyuod property=og:site_name><meta content=https://yo.eki.do/images/default.png data-svelte=svelte-zqyuod property=og:image> <link href=/client/client.ad92a8a1.js rel=modulepreload as=script crossorigin=use-credentials><link href=/client/client-0ed4f0b2.css rel=preload as=style><link href=/client/_slug_.907a21cd.js rel=modulepreload as=script crossorigin=use-credentials><link href=/client/utility.16d055df.js rel=modulepreload as=script crossorigin=use-credentials><link href=/client/inject_styles.803b7e80.js rel=modulepreload as=script crossorigin=use-credentials><link href=/client/_slug_-ad5942c7.css rel=preload as=style></head> <body> <div id=sapper> <nav class=svelte-1qy0d7s><ul class=svelte-1qy0d7s><li class=svelte-1qy0d7s><a href=/ class=svelte-1qy0d7s rel=prefetch>articles</a></li> <li class=svelte-1qy0d7s><a href=/about class=svelte-1qy0d7s rel=prefetch>about</a></ul> </nav> <main class=svelte-gph4rb> <header class=svelte-1tkryui><div class="svelte-1tkryui img-header" style=""></div> <h1 class=svelte-1tkryui>jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処</h1> <time class=svelte-1tkryui title=6 年前>2016-08-26<span class=date-weak>T</span>09:17:53<span class=date-weak>+09:00</span></time> <div class=tags></div> <p class="svelte-1tkryui history"><a href=https://github.com/kissge/yo.eki.do/commits/master/notes/jedi-virtualenv.md>更新履歴</a></header> <h1 class="svelte-1tkryui floating-header">jediとかvirtualenvとかがReason: image not foundと言ってぶっ壊れたときの対処</h1> <div class="svelte-1tkryui callout">この記事は以前別のところにあったものを移してきたものです。完璧に移植できておらず、表示が崩れているかもしれません。 </div> <div class="svelte-1tkryui content"><p>ある日突然jedi（EmacsでPythonを書くときにつかうすごいやつだよ）がぶっ壊れた。 python-modeのバッファでピリオドを打つたびに</p> <pre><code class=hljs>Error (jedi): Failed to start Jedi EPC server.
*** You may need to run "M-x jedi:install-server". ***
This could solve the problem especially if you haven't run the command yet
since Jedi.el installation or update and if the server complains about
Python module imports.
</code></pre> <p>というメッセージが出てくるし、当然補完機能は使えない。 指示通りコマンドを実行しても何も起きない。</p> <p>よくみると*Messages*に</p> <pre><code class=hljs>Server may raise an error. Use "M-x epc:pop-to-last-server-process-buffer RET" to see full traceback:
</code></pre> <p>とあるのでtracebackを見てみると</p> <pre><code class=hljs>dyld: Library not loaded: *****
  Referenced from: *****:
  Reason: image not found
</code></pre> <p>という見慣れないエラーがある。なんだこれ。</p> <h3>原因</h3> <p>簡単に言うとvirtualenvが壊れていた。 私の環境では <code>(custom-set-variables '(jedi:environment-root "jedi"))</code> という<a href=https://github.com/kissge/.emacs.d/blob/master/inits/15-python.el#L8>感じ</a>でvirtualenvを使う設定になっているのだが、そのディレクトリの中にある大量のシンボリックリンクがいずれもリンク切れになっていた。 homebrewでPythonをバージョンアップしたためである。</p> <p>と、いうわけで、このディレクトリを削除して作りなおすと万事うまくいった。よかった。</p> <p>あんまし関係無いけど、この問題に対処する隙がない間エラーを無視してとりあえずjediを <code>el-get-remove</code> しようとしたら <code>auto-complete</code> がないから無理〜と言って怒られたのは納得がいかない。 jediさん何とかして。</p> </div> <x-script async data-ad-client=ca-pub-5159408992513362 src=https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js></x-script> <section class="svelte-1tkryui mentions"><h1 class=svelte-1tkryui>Recent public mentions on Twitter</h1> <ol class=svelte-1tkryui reversed>Loading mentions...</ol> </section> </main></div> 